### ISSUE TITLE
JavaScript heap out of memory

### ISSUE THREAD

**Issue by bmino** [bmino/binance-triangle-arbitrage#2]
The program crashes with this error after running for a prolonged period of time:

```
<--- Last few GCs --->

[3570:0x3b24b80]  3363864 ms: Mark-sweep 1326.5 (1577.0) -> 1326.5 (1577.5) MB, 1142.6 / 0.1 ms  allocation failure GC in old space requested
[3570:0x3b24b80]  3364776 ms: Mark-sweep 1326.5 (1577.5) -> 1326.4 (1478.5) MB, 911.2 / 0.0 ms  last resort GC in old space requested
[3570:0x3b24b80]  3365795 ms: Mark-sweep 1326.4 (1478.5) -> 1326.4 (1446.5) MB, 1019.0 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1cb9778a55e9 <JSObject>

    1: depthData(aka depthData) [/home/brandon/github/binance-triangle-arbitrage/node_modules/node-binance-api/node-binance-api.js:~666] [pc=0x14a07df7cea8](this=0x510273822d1 <undefined>,data=0x2a92f1411ff1 <Object map = 0x21fa18bbf191>)
    2: /* anonymous */(aka /* anonymous */) [/home/brandon/github/binance-triangle-arbitrage/node_modules/node-binance-api/node-binance-api.js:1673] [bytecode=...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```

**Comment by bmino** [comment]
Internally there are two depth caches maintained. One is done by the third party module "node-binance-api" and another by this program. The cache maintained by this program is pruned to contain no more than 100 bids and 100 asks per symbol. However, the one maintained by "node-binance-api" is not pruned and will grow infinitely.

**Comment by bmino** [comment]
Removed the duplicate depthCache so the application only uses one which is appropriately pruned. However, the memory leak is still present and appears to be caused by the thread pool that performs the arbitrage calculations.

**Comment by bmino** [comment]
Fixed by bda8cf9fab395abfdbca7be0d8e7eebce5a7d7f3
